multi-filament/src/main/java/net/legacyfabric/multifilament/task/UnifyMappingsTask.java

```diff
@@ -0,0 +1,104 @@
+package net.legacyfabric.multifilament.task;
+
+import net.fabricmc.mappingio.MappingReader;
+import net.fabricmc.mappingio.MappingUtil;
+import net.fabricmc.mappingio.MappingWriter;
+import net.fabricmc.mappingio.format.MappingFormat;
+import net.fabricmc.mappingio.tree.MappingTree;
+import net.fabricmc.mappingio.tree.MemoryMappingTree;
+
+import net.legacyfabric.multifilament.mappingsio.FilteringMappingVisitor;
+
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.InputFile;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.BasicFileAttributes;
+
+public abstract class UnifyMappingsTask extends MappingOutputTask {
+	@Inject
+	public UnifyMappingsTask() {
+		this.getOutputFormat().convention(MappingFormat.ENIGMA);
+	}
+
+	@Input
+	public abstract Property<String> getUnifiedDir();
+
+	@InputDirectory
+	public abstract DirectoryProperty getVersionedDir();
+
+	void pre() throws IOException {
+		File file = new File(this.getVersionedDir().getAsFile().get().getParentFile(), this.getUnifiedDir().get());
+		file.delete();
+		Files.move(this.getOutputDir().getAsFile().get().toPath(), file.toPath());
+	}
+
+	void run(MappingWriter var1) throws IOException {
+		File file = new File(this.getVersionedDir().getAsFile().get().getParentFile(), this.getUnifiedDir().get());
+		MemoryMappingTree treeView = new MemoryMappingTree();
+		MappingReader.read(file.toPath(), treeView);
+		MemoryMappingTree mappingTree = new MemoryMappingTree();
+		MappingReader.read(this.getVersionedDir().getAsFile().get().toPath(), mappingTree);
+		mappingTree.accept(treeView);
+		treeView.accept(var1);
```

Comment:
MappingTrees aren't sorted by default:
https://github.com/FabricMC/mapping-io/blob/972ff892cc288428b9afdffd7c9cb751b1956119/src/main/java/net/fabricmc/mappingio/tree/MappingTreeView.java#L107-L109

This means you can get an inconsistent order after each mapping update, which makes things harder to review through unnecessarily larger diffs. We can easily sort the mappings in a consistent manner before writing them to disk by passing a `VisitOrder` as an additional argument:

```suggestion
		treeView.accept(var1, VisitOrder.createByName());
```

Reply:
Has been fixed in #269